src/sqlfluff/dialects/dialect_postgres.py

```diff
@@ -88,15 +87,62 @@
 
 @postgres_dialect.segment(replace=True)
 class FunctionDefinitionGrammar(BaseSegment):
-    """This is the body of a `CREATE FUNCTION AS` statement."""
+    """This is the body of a `CREATE FUNCTION AS` statement.
+
+    Options supported as defined in https://www.postgresql.org/docs/9.1/sql-createfunction.html
+    """
 
     match_grammar = Sequence(
-        "AS",
-        OneOf(Ref("QuotedLiteralSegment"), Ref("DollarQuotedLiteralSegment")),
+        AnyNumberOf(
+            Sequence("TRANSFORM", "FOR", "TYPE", Ref("ParameterNameSegment")),
+            OneOf("WINDOW", "IMMUTABLE", "STABLE", "VOLATILE", "STRICT"),
+            Sequence(OneOf("NOT", optional=True), "LEAKPROOF"),
+            Sequence(
+                OneOf("CALLED", Sequence("RETURNS", "NULL")), "ON", "NULL", "INPUT"
+            ),
+            Sequence(
+                OneOf("EXTERNAL", optional=True),
+                "SECURITY",
+                OneOf("INVOKER", "DEFINER"),
+            ),
+            Sequence("LANGUAGE", Ref("ParameterNameSegment")),
```

Comment:
Any reason not to have this at the top as per the spec?

Reply:
Fixed